packages/core/src/Provider/TxSubmitProvider/types.ts

```diff
@@ -14,5 +14,5 @@ export interface TxSubmitProvider extends Provider {
    * @param signedTransaction signed and serialized cbor
    * @throws {CardanoNodeErrors.TxSubmissionError}
    */
-  submitTx: (args: SubmitTxArgs) => Promise<void>;
+  submitTx: (args: SubmitTxArgs, handleProvider?: HandleProvider) => Promise<void>;
```

Comment:
Let's not change the signature of this, because this interface is used in different contexts (client-side as well). HandleProvider is an implementation detail of a specific TxSubmitProvider
```suggestion
  submitTx: (args: SubmitTxArgs) => Promise<void>;
```

Reply:
0dfd234d683942a8f490be4b630f098e8d0fa823 should solve this